### ISSUE TITLE
[BUG] CPU OOM During Training, Possibly Caused by `manual_gc`

### ISSUE THREAD

**Issue by lhcstation** [nvidia-cosmos/cosmos-transfer2.5#168]
**Describe the bug**

Hi, thanks for the awesome work!

I encountered a memory leak/crash when fine-tuning the single-view model on my own dataset (similar to DEFAULT_BASE_EXPERIMENT).

The training is stable for the first ~1200 iterations, but then there was a significant increase in cpu memory usage, and the training crashed (see image below).

The issue appears to be related to the dataloader, specificly the `manual_gc` callback.

I tried to disable the manual_gc callback, and training runs stably without crashes.

I believe the issue is now clear:
- In the manual_gc callback, after a warmup period (1000 iters for default), `auto gc` is turned off and switched to `manual gc` to reduce GPU sync time. 

- After switched to `manual gc`, the cpu memory abnormal increase and the training soon collapsed.

- Disabling the manual_gc callback, which means keep `manual gc` enabled throughout training, resolves the issue.

This issue is likely hardware- or environment-specific on my setup.
However, disabling manual_gc ensures stable training, and this workaround may be helpful for others encountering similar issue.

**To Reproduce**
N/A

**Environment (please complete the following information):**
 - Revision: e899a2f
 - Environment: uv
 - Hardware: H200
 - CUDA Version: 12.8 (570.124.06)

<img width="683" height="279" alt="Image" src="https://github.com/user-attachments/assets/91a56f73-106d-416e-a2bc-466dec6db478" />
